mslib/mswms/gallery_builder.py

```diff
@@ -0,0 +1,523 @@
+    </a>""" if link else f"""<img src="{image_location}" style="width: 100 % "/>"""
+    return f"""<div class="gallery">
+                 {image}
+                 <div class="gtooltip">
+                  {caption}<span class="gtooltiptext">{tooltip}</span>
+                 </div>
+                </div>"""
+
+
+def write_doc_index():
+    """
+    Write index containing all code examples for the sphinx docs
+    """
+    with open(os.path.join(DOCS_LOCATION, "code", "index.rst"), "w+") as rst:
+        files = "\n".join(sorted(["   " + f[:-4] for f in os.listdir(os.path.join(DOCS_LOCATION, "code"))
+                                  if "index" not in f and ".rst" in f]))
+        rst.write(f"""
+Code Examples
+--------------
+
+.. toctree::
+
+{files}
+""")
+
+
+def write_html(sphinx=False):
+    """
+    Writes the plots.html file containing the gallery
+    """
+    location = DOCS_LOCATION if sphinx else STATIC_LOCATION
+    html = begin
+    if sphinx:
+        html = html.replace("<h3>Plot Gallery</h3>", "")
+
+    for l_type in plots:
+        style = ""
+        if l_type == "Top":
+            style = "style=\"display: block;\""
+        html += f"<div id=\"{l_type}-View\" class=\"tabcontent\" {style}>"
+        html += "\n".join(plots[l_type])
+        html += "</div>"
+
+    with open(os.path.join(location, "plots.html"), "w+") as file:
+        file.write(html + end)
+        logging.info(f"plots.html created at {os.path.join(location, 'plots.html')}")
+
+
+def import_instructions(plot_object, l_type, layer, native_import=None, dataset=""):
+    """
+    Returns instructions on how to import the plot object, or None if not yet implemented
+    """
+    # Imports here due to some circular import issue if imported too soon
+    from mslib.mswms.mpl_lsec_styles import LS_DefaultStyle
+    from mslib.mswms.mpl_vsec_styles import VS_GenericStyle
+    from mslib.mswms.mpl_hsec_styles import HS_GenericStyle
+
+    from_text = f"{l_type}_{dataset}{plot_object.name}" if not native_import else native_import
+    instruction = f"from {from_text} import {plot_object.__class__.__name__}\n" \
+                  f"register_{layer}_layers = [] if not register_{layer}_layers else register_{layer}_layers\n"
+    # Generic LS class
+    if isinstance(plot_object, LS_DefaultStyle):
+        instruction += f"register_{layer}_layers.append(({plot_object.__class__.__name__}, " \
+                       f"\"{plot_object.required_datafields[-1][1]}\", \"{plot_object.required_datafields[-1][0]}\", " \
+                       f"[next(iter(data))]))"
+    # Generic HS or VS class without custom _prepare_datafields
+    elif (isinstance(plot_object, VS_GenericStyle) or isinstance(plot_object, HS_GenericStyle)) and "pass" in \
+            inspect.getsource(plot_object._prepare_datafields):
+        style = "hsec" if isinstance(plot_object, HS_GenericStyle) else "vsec"
+        if plot_object.contours:
+            for i in range(len(plot_object.contours)):
```

Comment:
`
for i, temp in enumerate(plot_object.contours):
`
?

Reply:
Why is this even necessary? Should a repr() below in the string deal with this?
We could also modify the class to provide better instructions. It is our "go-to" class for generic plotting and should be improved to handle more use cases in the future.

Reply:
Having any numpy array and representing it as a string will yield an unparseable syntax like "array([1, 2, 3])" or "[1 2 3]".
Converting it to a normal list beforehand makes sure the class can be used as is.

Reply:
Mmh. I see the problem. But the intent of the code is not obvious. In this case a comment w.r.t. to the intent (replacement of numpy arrays) would be helpful. Maybe move the conversion towards the make_generic functions (with a helper). The gallery code shows the need for better self-documentation/introspection of that classes.